### PR TITLE
Ruby on Rails capstone project: Budget app

### PR DESCRIPTION
**Pull request by anasbassi**
In this project the following requirements were implemented:
- register and log in, so that the data is private to them.
- introduce new transactions associated with a category.
- see the money spent on each category.
- I also created unit and integration tests for all the most important components of RoR application.
click [here](https://www.loom.com/share/2917e0e8d3cd40faa23cdba6f7de67ef) to view a description of the project

### CODE REVIEW

File: app/views/layouts/_navigation.html.erb [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+          <%= link_to 'Budget App', root_path, class: "nav-link" %>
+        </li>
+        <li class="nav-item">
+          <%= link_to 'Categories', categories_path, class: "nav-link" %>
+        </li>
+        <li class="nav-item">
+          <% if user_signed_in? %>
+            <%= link_to "Logout", destroy_user_session_path, method: :delete, :class => 'nav-link' %>
+          <% else %>
+            <%= link_to "Login", new_user_session_path, :class => 'nav-link' %>
```
Comment (arslanbisharat) [inline app/views/layouts/_navigation.html.erb:18]: - [ ] When the user logs in, they should be presented with the categories page.

File: app/views/layouts/_navigation.html.erb [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+      <span class="navbar-toggler-icon"></span>
+    </button>
+    <div class="collapse navbar-collapse" id="navbarSupportedContent">
+      <ul class="navbar-nav me-auto mb-2 mb-lg-0">
+        <li class="nav-item">
+          <%= link_to 'Budget App', root_path, class: "nav-link" %>
+        </li>
+        <li class="nav-item">
+          <%= link_to 'Categories', categories_path, class: "nav-link" %>
+        </li>
```
Comment (arslanbisharat) [inline app/views/layouts/_navigation.html.erb:13]: - [ ] There should be a back button `>` in your design. When the user clicks on the "Back" button (<), the user navigates to the home page

File: app/views/layouts/_navigation.html.erb [hunk 3]
```diff
@@ -0,0 +1,29 @@
+<nav class="navbar navbar-expand-lg navbar-dark bg-primary">
+  <div class="container-fluid">
+    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarSupportedContent" aria-controls="navbarSupportedContent" aria-expanded="false" aria-label="Toggle navigation">
+      <span class="navbar-toggler-icon"></span>
+    </button>
+    <div class="collapse navbar-collapse" id="navbarSupportedContent">
+      <ul class="navbar-nav me-auto mb-2 mb-lg-0">
+        <li class="nav-item">
+          <%= link_to 'Budget App', root_path, class: "nav-link" %>
```
Comment (arslanbisharat) [inline app/views/layouts/_navigation.html.erb:9]: - [ ] Kindly make sure you are using the design color scheme and layout. Currently, your app doesn't follow it.

File: test/channels/application_cable/connection_test.rb [hunk 4]
```diff
@@ -1,4 +1,4 @@
-require "test_helper"
+require 'test_helper'
```
Comment (arslanbisharat) [inline test/channels/application_cable/connection_test.rb:1]: - [ ] Please make sure App has unit tests for all the most important components of the RoR application.

**Review by arslanbisharat — CHANGES_REQUESTED** [review]
Hi @anasbassi & @Aquinasmoriasi, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights

- The Pull Request has a proper title and description. ✔️
- You have added a descriptive Readme file. Good Job. ✔️
- There is no linter error is present. ✔️
- Project has met all the technical aspects you needed for this project. Keep rocking! ✔️

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- [ ] Please make sure you right descriptive comments. Commenting involves placing Human Readable Descriptions inside of computer programs detailing what the Code is doing. Proper use of commenting can make code maintenance much easier, as well as helping make finding bugs faster. Further, commenting is very important when writing functions that other people will use. :x:


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: test/channels/application_cable/connection_test.rb (hunk 4 above)
Comment (anasbassi) [reply test/channels/application_cable/connection_test.rb:1]: All important unit tests were written and all are passing
rspec-pass

**Review by Deepakdanger — APPROVED** [review]
Hi @anasbassi , 

### Status: Approved ✔️

Your project is complete!  There is nothing else to say other than... it's time to merge it  🚢 
Congratulations! 🎉 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- x

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by CollinsTatang — APPROVED** [review]
# Approved 👏  
### **Highlights 🚀 :heavy_check_mark: :heavy_check_mark:** 

- The repo looks professional.:heavy_check_mark:
- Git/GitHub flow best practices have been followed:heavy_check_mark:
- You have descriptive PR:heavy_check_mark:

In my opinion, your project is ready for presentation!  There is nothing else to say other than... it's time to merge it :shipit: 
### **Congratulations!** 🎉 
image



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag @CollinsTatang in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**